Patch-Source: https://salsa.debian.org/debian/clipit/-/blob/debian/1.4.5+git20210313-3/debian/patches/incompatible-pointer-types.patch

Author: Andreas Beckmann <anbe@debian.org>
Description: fix building with -Werror=incompatible-pointer-types

--- a/src/keybinder.c
+++ b/src/keybinder.c
@@ -310,7 +310,7 @@ keybinder_is_modifier (guint keycode)
 	gboolean retval = FALSE;
 
 	mod_keymap = XGetModifierMapping (
-	    GDK_WINDOW_XDISPLAY(gdk_display_get_default()));
+	    GDK_DISPLAY_XDISPLAY(gdk_display_get_default()));
 
 	map_size = 8 * mod_keymap->max_keypermod;
 
--- a/src/main.c
+++ b/src/main.c
@@ -614,7 +614,7 @@ void underline_match(char* match, GtkMen
   PangoAttrList* attr_list = pango_attr_list_new();
   pango_attr_list_insert (attr_list, underline);
 
-  GtkWidget* gtk_label = gtk_bin_get_child (GTK_BIN (menu_item));
+  GtkLabel* gtk_label = (GtkLabel *) gtk_bin_get_child (GTK_BIN (menu_item));
   gtk_label_set_attributes (gtk_label, attr_list);
 }
 
@@ -638,7 +638,7 @@ gboolean selected_by_input(const GtkWidg
     append_to_input_buffer(event->string);
 
   GtkMenuShell* menu = (GtkMenuShell *) history_menu;
-  GList* element = gtk_container_get_children(menu);
+  GList* element = gtk_container_get_children((GtkContainer *) menu);
   GtkMenuItem *menu_item, *first_match = 0;
 
   const gchar* menu_label;
@@ -657,9 +657,9 @@ gboolean selected_by_input(const GtkWidg
         first_match = menu_item;
       match_count++;
       underline_match(match, menu_item, menu_label);
-      gtk_widget_set_sensitive(menu_item, true);
+      gtk_widget_set_sensitive((GtkWidget *) menu_item, true);
     } else {
-      gtk_widget_set_sensitive(menu_item, false);
+      gtk_widget_set_sensitive((GtkWidget *) menu_item, false);
     }
     element = element->next;
     count++;
@@ -667,7 +667,7 @@ gboolean selected_by_input(const GtkWidg
 
   if (first_match && match_count != prefs.items_menu) {
     gtk_menu_item_select(first_match);
-	gtk_menu_shell_select_item(menu, first_match);
+	gtk_menu_shell_select_item(menu, (GtkWidget *) first_match);
     return TRUE;
   }
   return FALSE;
--- a/src/preferences.c
+++ b/src/preferences.c
@@ -723,7 +723,7 @@ void show_preferences(gint tab) {
             *alignment, *hbox,
             *vbox;
 
-  GtkWidget *adjustment, *adjustment_small, *adjustment_statics;
+  GtkAdjustment *adjustment, *adjustment_small, *adjustment_statics;
   GtkTreeViewColumn *tree_column;
 
   /* Create the dialog */
@@ -898,9 +898,9 @@ void show_preferences(gint tab) {
   gtk_misc_set_alignment((GtkMisc*)label, 0.0, 0.50);
   gtk_box_pack_start((GtkBox*)hbox, label, FALSE, FALSE, 0);
   ellipsize_combo = gtk_combo_box_text_new();
-  gtk_combo_box_text_append_text((GtkComboBox*)ellipsize_combo, _("Beginning"));
-  gtk_combo_box_text_append_text((GtkComboBox*)ellipsize_combo, _("Middle"));
-  gtk_combo_box_text_append_text((GtkComboBox*)ellipsize_combo, _("End"));
+  gtk_combo_box_text_append_text((GtkComboBoxText*)ellipsize_combo, _("Beginning"));
+  gtk_combo_box_text_append_text((GtkComboBoxText*)ellipsize_combo, _("Middle"));
+  gtk_combo_box_text_append_text((GtkComboBoxText*)ellipsize_combo, _("End"));
   gtk_box_pack_start((GtkBox*)hbox, ellipsize_combo, FALSE, FALSE, 0);
   gtk_box_pack_start((GtkBox*)vbox_history, frame, FALSE, FALSE, 0);
 
